### PR TITLE
Use towncrier and improve setting dev version

### PR DESCRIPTION
**Pull request by ericpre**
Similar as https://github.com/hyperspy/holospy/pull/17

### Progress of the PR
- [x] Improve setting development version,
- [x] Use towncrier to manage release notes,
- [n/a] docstring updated (if appropriate),
- [x] update user guide (if appropriate),
- [x] added tests,
- [ ] added line to CHANGES.rst,
- [ ] ready for review.

### CODE REVIEW

File: exspy/__init__.py [hunk 1]
```diff
@@ -8,16 +27,29 @@
...
+_root = Path(__file__).resolve().parents[1]
+if (_root / ".git").exists() and not (_root / ".git/shallow").exists():
+    try:
+        # setuptools_scm may not be installed
+        from setuptools_scm import get_version
 
-    __version__ = get_version(Path(__file__).parent.parent)
+        __version__ = get_version(_root)
+    except ImportError:  # pragma: no cover
+        # setuptools_scm not install, we keep the existing __version__
```
Comment (jlaehne) [inline exspy/__init__.py:51]:
```suggestion
        # setuptools_scm not installed, we keep the existing __version__
```

File: exspy/__init__.py [hunk 2]
```diff
@@ -8,16 +27,29 @@
...
-    # Tested in the "build" workflow on GitHub CI
-    from importlib.metadata import version
+__version__ = version("exspy")
+
+# For development version, `setuptools_scm` will be used at build time
+# to get the dev version, in case of missing vcs information (git archive,
+# shallow repository), the fallback version defined in pyproject.toml will
+# be used
+
+# if we have a editable install from a git repository try to use
```
Comment (jlaehne) [inline exspy/__init__.py:37]:
```suggestion
# If we have an editable install from a git repository try to use
```

**Review by jlaehne — APPROVED** [review]

**Review by jlaehne — COMMENTED** [review]
Amend `PULL_REQUEST_TEMPLATE.md` as for holospy?

**Comment by ericpre** [comment]
> Amend `PULL_REQUEST_TEMPLATE.md` as for holospy?

Done!

**Pull request merged**